catroid/src/main/java/org/catrobat/catroid/utils/Utils.java

```diff
@@ -1104,4 +1104,14 @@ public static boolean isUserLoggedIn(Context context) {
 	public static String getNumberStringForBricks(float value) {
 		return (int) value == value ? "" + (int) value : "" + value;
 	}
+
+	public static <T extends Object> List<T> distinctListByClassOfObjects(List<T> listToDistinct) {
+		List<T> distinctList = new ArrayList<>();
+		HashMap<Class, T> uniqueMap = new HashMap<>();
+		for (T objectInstance : listToDistinct) {
+			uniqueMap.put(objectInstance.getClass(), objectInstance);
+		}
+		distinctList.addAll(uniqueMap.values());
+		return distinctList;
```

Comment:
> extends Object is necessary (IMHO) because i call Object.getClass(). Would not be able to do that (accourding to this question ).

Just give it a try by writing a few tests. :wink:
<details>
<summary>After that, expand me!</summary>
As you already know, for <a href="http://stackoverflow.com/questions/339699/java-generics-type-erasure-when-and-what-happens">some reason</a> T.class isn't valid. However, <code>T.class != Object.getClass()</code>. You can always call getClass() on any Java Object. Even on <code>String.class.getClass()</code> (hopefully this won't be confusing :D String.class is just the class object of the String). You just can't use .class with generics.<br>
If not specified otherwise, every Class extends from Object. So <code>class A {…}</code> == <code>class A extends Object {…}</code>. In java <code>extends Object</code> is (almost) always redundant.
The <a href="https://github.com/openjdk-mirror/jdk7u-jdk/blob/f4d80957e89a19a29bb9f9807d2a28351ed7f7df/src/share/classes/java/util/Collections.java#L215">jdk</a> and <a href="https://github.com/google/guava/blob/8a991157845af123f701f223c8f2ecc245e82977/guava/src/com/google/common/collect/Lists.java#L111">guava</a> also do not use extends Object.
</details>

</br>

> Map as declaration for what exactly?
```java
Map<Class, T> uniqueMap = new HashMap<>();
````